src/Microsoft.AspNetCore.ResponseCaching/Internal/MemoryResponseCacheStore.cs

```diff
@@ -3,46 +3,86 @@
 
 using System;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
 using Microsoft.Extensions.Caching.Memory;
-using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.AspNetCore.ResponseCaching.Internal
 {
     public class MemoryResponseCacheStore : IResponseCacheStore
     {
         private readonly IMemoryCache _cache;
+        private readonly ResponseCacheOptions _options;
 
-        public MemoryResponseCacheStore(IMemoryCache cache)
+        public MemoryResponseCacheStore(IMemoryCache cache, IOptions<ResponseCacheOptions> options)
         {
             if (cache == null)
             {
                 throw new ArgumentNullException(nameof(cache));
             }
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
 
             _cache = cache;
+            _options = options.Value;
```

Comment:
Same option in `ResponseCacheOptions` configures the shard size of the buffer in `ResponseCacheStream` and `ReadOnlyShardStream`.